### PR TITLE
chore: Enable more linters

### PR DESCRIPTION
**Pull request by k15r**
## Description

Changes proposed in this pull request (what was done and why):

- enable intrange linter
- enable noctx linter
- enable makezero linter
- enable gocritic linter
- enable mnd linter
- enable exhaustive linter
- enable wsl linter
- add wsl to autofix make target

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: apis/telemetry/v1beta1/logpipeline_validation.go [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+	"errors"
 	"fmt"
 	"regexp"
 	"strings"
 
 	"github.com/kyma-project/telemetry-manager/internal/fluentbit/config"
 )
 
+var (
+	validHostNameRE              = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
```
Comment (skhalash) [inline apis/telemetry/v1beta1/logpipeline_validation.go:13]: Please use the same name in v1alpha1 and v1beta1 validation for consistency.

File: internal/fluentbit/config/builder/config_builder.go [hunk 2]
```diff
@@ -8,6 +8,10 @@ import (
 	"github.com/kyma-project/telemetry-manager/internal/fluentbit/config"
 )
 
+var (
+	ErrUndefinedOutputPlugin = fmt.Errorf("output plugin not defined")
```
Comment (skhalash) [inline internal/fluentbit/config/builder/config_builder.go:12]:
```suggestion
	ErrUndefinedOutputPlugin = errors.New("output plugin not defined")
```

File: internal/tools/tools.go [hunk 3]
```diff
@@ -18,4 +18,5 @@ import (
 	_ "golang.org/x/tools/cmd/stringer"
 	_ "sigs.k8s.io/controller-tools/cmd/controller-gen"
 	_ "sigs.k8s.io/kustomize/kustomize/v5"
+    _ "github.com/bombsimon/wsl/v4/cmd/wsl"
```
Comment (skhalash) [inline internal/tools/tools.go:21]: Wrong indentation?

**Review by skhalash — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -75,14 +76,17 @@ help: ## Display this help.
...
+	-$(WSL) --fix ./...
 	$(GOLANGCI_LINT) run --fix
 
 lint: $(GOLANGCI_LINT)
 	go version
 	$(GOLANGCI_LINT) version
 	GO111MODULE=on $(GOLANGCI_LINT) run
 
+	$(GOLANGCI_LINT) run --fix
+
```
[the remark this answers is not in the document]
Comment (k15r) [reply Makefile:89]: Nope

File: apis/telemetry/v1beta1/logpipeline_validation.go (hunk 1 above)
Comment (k15r) [reply apis/telemetry/v1beta1/logpipeline_validation.go:13]: good catch

File: internal/tools/tools.go (hunk 3 above)
Comment (k15r) [reply internal/tools/tools.go:21]: not according to my IDE

Comment (k15r) [reply internal/tools/tools.go:21]: but wrong order :D

**Review by skhalash — APPROVED** [review]

**Pull request merged**